lightwood/encoders/text/pretrained.py

```diff
@@ -0,0 +1,323 @@
+                self._pretrained_model_name,
+                num_labels=label_size,
+            ).to(self.device)
+
+            # Construct the dataset for training
+            xinp = TextEmbed(text, labels)
+            dataset = DataLoader(xinp, batch_size=self._batch_size, shuffle=True)
+
+            # If max length not set, adjust
+            if self._max_len is None:
+                self._max_len = self._model.config.max_position_embeddings
+
+            if self._frozen:
+                print("\tFrozen Model + Training Classifier Layers", flush=True)
+                """
+                Freeze the base transformer model and train
+                a linear layer on top
+                """
+                # Freeze all the transformer parameters
+                for param in self._model.base_model.parameters():
+                    param.requires_grad = False
+
+                optimizer_grouped_parameters = self._model.parameters()
+
+            else:
+                print("\tFine-tuning model", flush=True)
+                """
+                Fine-tuning parameters with weight decay
+                """
+                no_decay = [
+                    "bias",
+                    "LayerNorm.weight",
+                ]  # decay on all terms EXCLUDING bias/layernorms
+                optimizer_grouped_parameters = [
+                    {
+                        "params": [
+                            p
+                            for n, p in self._model.named_parameters()
+                            if not any(nd in n for nd in no_decay)
+                        ],
+                        "weight_decay": 0.01,
+                    },
+                    {
+                        "params": [
+                            p
+                            for n, p in self._model.named_parameters()
+                            if any(nd in n for nd in no_decay)
+                        ],
+                        "weight_decay": 0.0,
+                    },
+                ]
+
+            optimizer = AdamW(optimizer_grouped_parameters, lr=1e-5)
+            scheduler = get_linear_schedule_with_warmup(
+                optimizer,
+                num_warmup_steps=0,  # default value for GLUE
+                num_training_steps=len(dataset) * self._epochs,
+            )
+
+            # Train model; declare optimizer earlier if desired.
+            self._train_model(
+                dataset, optim=optimizer, scheduler=scheduler, n_epochs=self._epochs
+            )
+
+        else:
+            print("Embeddings Generator only", flush=True)
+
+            self.model_type = "embeddings_generator"
+            self._model = self._embeddings_model_class.from_pretrained(
+                self._pretrained_model_name
+            ).to(self.device)
+
+        self._prepared = True
+
+    def _train_model(self, dataset, optim=None, scheduler=None, n_epochs=1):
```

Comment:
Please avoid superfolous arguments. If they aren't ever used, don't put them, if they are, don't have a default value.
We should limit arguments with default values to interfaces which are used in a lot of places, this is only used once, so it makes on sense to have the defaults